src/realm/object-store/sync/app.cpp

```diff
@@ -777,6 +778,51 @@ void App::remove_user(std::shared_ptr<SyncUser> user, std::function<void(Optiona
     }
 }
 
+void App::delete_user(std::shared_ptr<SyncUser> user, util::FunctionRef<void(Optional<AppError>)> completion_block)
+{
+    if (!user || user->state() == SyncUser::State::Removed) {
```

Comment:
From the tests, it appears that this works with a logged out user and that is surprising to me. I would expect that deleting a user requires the user to be logged in. If the user is logged out, the refresh token used in the below request would be empty. If the server actually lets anyone delete users without being authenticated, that seems like a security issue we should push back on?

Reply:
My bad,I was not using the integration test config, so it would always pass. I'll change it